$DragonFly: src/usr.bin/awk/patches/tran.c.patch,v 1.1 2004/11/09 08:47:36 joerg Exp $

Index: /tran.c
===================================================================
RCS file: /home/dcvs/src/contrib/awk20040207/tran.c,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 tran.c
--- tran.c	17 Apr 2004 19:41:31 -0000	1.1.1.1
+++ tran.c	31 Oct 2004 03:18:48 -0000
@@ -397,10 +397,9 @@ char *tostring(const char *s)	/* make a 
 {
 	char *p;
 
-	p = (char *) malloc(strlen(s)+1);
+	p = strdup(s);
 	if (p == NULL)
 		FATAL("out of space in tostring on %s", s);
-	strcpy(p, s);
 	return(p);
 }
 
